bloom-filter/src/main/scala/bloomfilter/mutable/BloomFilter.scala

```diff
@@ -12,17 +12,27 @@ class BloomFilter[T] private (val numberOfBits: Long, val numberOfHashes: Int, p
     this(numberOfBits, numberOfHashes, new UnsafeBitArray(numberOfBits))
   }
 
+ /** Add an item in the Bloom filter
+   *
+   * @param x the T item to add
+   * @return true if the item was not in the filter before, false otherwise
+   */
   def add(x: T): Unit = {
     val hash = canGenerateHash.generateHash(x)
     val hash1 = hash >>> 32
     val hash2 = (hash << 32) >> 32
+    var was_defined = true
 
     var i = 0
     while (i < numberOfHashes) {
       val computedHash = hash1 + i * hash2
+      if (!bits.get((computedHash & Long.MaxValue) % numberOfBits))
```

Comment:
I think the best way is a separate method in `BloomFilter` class and another one in `UnsafeBitArray` (it has the read operation there already)

```
UnsafeBitArray{
  def setAndGet?(index: Long): Boolean = {
    val offset = ptr + (index >>> 6) * 8L
    val long = unsafe.getLong(offset)
    if (bit not set in long) {
        unsafe.putLong(offset, long | (1L << index))
        true
    } else false
  }
}
```